server/src/clj_kondo/lsp_server/impl/server.clj

```diff
@@ -114,14 +115,20 @@
                 (.getParentFile))]
     (config-dir dir)))
 
+(defn config-options
+  [args path]
+  (str/join "/" (concat [path] [(first args)])))
+
 (defn lint! [text uri]
+  
   (let [lang (uri->lang uri)
-        cfg-dir (uri->config-dir uri)
+        ^java.io.File cfg-dir (uri->config-dir uri)
         {:keys [:findings]} (with-in-str text
                               (clj-kondo/run! (cond->
                                                   {:lint ["-"]
                                                    :lang lang}
-                                                cfg-dir (assoc :config-dir cfg-dir))))
+                                                cfg-dir (assoc :config-dir cfg-dir)
+                                                cfg-dir (assoc :config (config-options @arg-options (.getPath cfg-dir))))))
```

Comment:
- [ ] What if there is no config directory and we still want to pass the config to clj-kondo?

Reply:
Will make this conditional.